pkg/cmd/release/create/create.go

```diff
@@ -435,8 +435,16 @@ func createRun(opts *CreateOptions) error {
 
 	hasAssets := len(opts.Assets) > 0
 
-	// Avoid publishing the release until all assets have finished uploading
-	if hasAssets {
+	if hasAssets && !opts.Draft {
+		// Check for an existing release
+		if opts.TagName != "" {
+			if ok, err := publishedReleaseExists(httpClient, baseRepo, opts.TagName); err != nil {
```

Comment:
Is there a good reason to not do this check for all cases rather than just this one? Seems like it would simplify the flow a bit if we always check if a release exists before proceeding.

Reply:
That's a good question. In all other cases, server-side validation would provide a sufficient check for whether a release with the same tag name exist. In those cases, repeating the same check on the client is not just redundant, but also sensitive to race conditions: a duplicate release may have been created in the short time between the check happening and the create release operation.